server/src/main/java/io/crate/planner/operators/HashJoin.java

```diff
@@ -129,11 +121,9 @@ public ExecutionPlan build(PlannerContext plannerContext,
         LogicalPlan leftLogicalPlan = lhs;
         LogicalPlan rightLogicalPlan = rhs;
 
-        boolean tablesSwitched = false;
```

Comment:
There is no need for a table-switched flag anymore. The hash-symbols are calculated once the lhs/rhs operators are already flipped.